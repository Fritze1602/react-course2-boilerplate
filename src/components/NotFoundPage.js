import React from 'react'

import {Link} from 'react-router-dom'

const NotFoundPage = () => (
    <div>
        <p>404 This Page can't be found</p>
       <Link to="/">Go Home</Link>
    </div>
);

export default NotFoundPage;
