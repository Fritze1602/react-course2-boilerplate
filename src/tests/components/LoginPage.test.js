import React from 'react';
import {shallow} from 'enzyme';
import {LoginPage} from '../../components/LoginPage';

let wrapper, startLogin;

beforeEach(()=>{
    startLogin = jest.fn();
    wrapper = shallow(<LoginPage startLogin={startLogin} /> );
})

test('Should render Login Page correctly', ()=>{
    expect(wrapper).toMatchSnapshot();
})

test('Should call start Login on button lick', ()=>{
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
})