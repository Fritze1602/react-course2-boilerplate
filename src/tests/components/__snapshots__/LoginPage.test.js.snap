// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Login Page correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Boilerplate React App
    </h1>
    <p>
      Tagline for App
    </p>
    <button
      className="button"
      onClick={[MockFunction]}
    >
      Login with Google
    </button>
  </div>
</div>
`;
